### PR TITLE
Add a handful of integration tests

### PR DESCRIPTION
**Pull request by BackEndTea**

### CODE REVIEW

**Comment by keradus** [comment]
Hi. please rebase on 2.2

**Comment by BackEndTea** [comment]
Targeted 2.2 in #3440. 
I made it a separate PR, since not all fixers here are present in 2.2

File: tests/Fixtures/Integration/priority/class_attributes_separation,braces.test [hunk 1]
```diff
@@ -0,0 +1,25 @@
+--TEST--
```
Comment (SpacePossum) [inline tests/Fixtures/Integration/priority/class_attributes_separation,braces.test]: Hi @BackEndTea thanks for picking this up!
I've trying out this integration-priority-test. The test passes which is good of course, however if I change the priority the test passes as well. For example I've changed the priority for the ClassAttributesSeparationFixer to 5500 and -5500.
This implies that this test proves the fixers work well together, but not that there is a priority issue. This last conclusion is unfortunate as that is why we write these test. Could you double-check this test?

Comment (keradus) [reply tests/Fixtures/Integration/priority/class_attributes_separation,braces.test]: So...
I did double check.
This itests indeed shall be moved to `misc` tests, it doesn't belong to `priority` tests.
We do have check that make sure if fixers would be applied in reverse order the expected outcome must be the same. But we don't check if order matters for `priority` cases.

@BackEndTea , please move this itest to `misc` or extend it to prove it is a priority tests indeed.

@SpacePossum , let us move discussion to: #3444

**Comment by BackEndTea** [comment]
@keradus @SpacePossum I have moved the discussed test to the misc folder for now, and added it back to the list of caseswithout tests

**Comment by keradus** [comment]
#3440 was merged, please rebase

**Comment by BackEndTea** [comment]
Rebased and targeted 2.10

File: tests/Fixtures/Integration/misc/class_attributes_separation,braces.test [hunk 2]
```diff
@@ -0,0 +1,25 @@
+--TEST--
```
Comment (keradus) [inline tests/Fixtures/Integration/misc/class_attributes_separation,braces.test]: now I got lost, this one is in `misc`, and on the other side it's listed as missing `prio` test

Comment (BackEndTea) [reply tests/Fixtures/Integration/misc/class_attributes_separation,braces.test]: My bad, if removed it from the list of to check priority checkers, and removed it as missing a test

**Comment by keradus** [comment]
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3444 got merged, I merged 2.10 into your PR.
let us see...

**Comment by BackEndTea** [comment]
@keradus It looks like the appveyor failures are unrelated to my changes.

**Comment by SpacePossum** [comment]
@BackEndTea please note that I've resolved the merge conflict, so please update your local branch if needed

**Comment by keradus** [comment]
Let us not merge this one yet, we got no answer for `class_attributes_separation,braces.test`, and for that I want to manually test it

**Comment by SpacePossum** [comment]
sure @keradus , I couldn't find a conflict but like hear your findings :)

**Comment by keradus** [comment]
maybe i won't find a thing, simply need to find an hour of calm and double check it

**Comment by SpacePossum** [comment]
I found a prio issue after all and created the test.
Also updated the fixer factory for it, rebased, resolved conflicts and squashed.
I know people don't like early squashing but maintaining a PR for 10+ months causes this type of thing.

**Comment by keradus** [comment]
> I know people don't like early squashing (...)

next time, simply keep single commit for original author's changes, and different commit for your changes - now, impossible to see what was your fix

> I found a prio issue after all and created the test.
> maintaining a PR for 10+

ha, I knew we had a prio issue somewhere here ! nice finding !

**Comment by SpacePossum** [comment]
Thank you @BackEndTea.

**Comment by BackEndTea** [comment]
Thank you @SpacePossum & @keradus

**Pull request merged**